src/app/WriteHandler.h

```diff
@@ -0,0 +1,135 @@
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *     This file defines write handler for a CHIP Interaction Data model
+ *
+ */
+
+#pragma once
+#include <app/AttributePathParams.h>
+#include <app/InteractionModelDelegate.h>
+#include <app/MessageDef/WriteResponse.h>
+#include <core/CHIPCore.h>
+#include <core/CHIPTLVDebug.hpp>
+#include <messaging/ExchangeContext.h>
+#include <messaging/ExchangeMgr.h>
+#include <messaging/Flags.h>
+#include <protocols/Protocols.h>
+#include <support/CodeUtils.h>
+#include <support/DLLUtil.h>
+#include <support/logging/CHIPLogging.h>
+#include <system/SystemPacketBuffer.h>
+
+namespace chip {
+namespace app {
+/**
+ *  @class WriteHandler
+ *
+ *  @brief The write handler is responsible for processing a write request and sending a write reply.
+ *
+ */
+class WriteHandler
+{
+public:
+    /**
+     *  Initialize the WriteHandler. Within the lifetime
+     *  of this instance, this method is invoked once after object
+     *  construction until a call to Shutdown is made to terminate the
+     *  instance.
+     *
+     *  @param[in]    apDelegate       InteractionModelDelegate set by application.
+     *
+     *  @retval #CHIP_ERROR_INCORRECT_STATE If the state is not equal to
+     *          kState_NotInitialized.
+     *  @retval #CHIP_NO_ERROR On success.
+     *
+     */
+    CHIP_ERROR Init(InteractionModelDelegate * apDelegate);
+
+    /**
+     *  Shut down the ReadHandler. This terminates this instance
+     *  of the object and releases all held resources.
+     *
+     */
+    void Shutdown();
+    /**
+     *  Process a write request.  Parts of the processing may end up being asynchronous, but the WriteHandler
+     *  guarantees that it will call Shutdown on itself when processing is done (including if OnWriteHandler
+     *  returns an error).
+     *
+     *  @param[in]    apExchangeContext    A pointer to the ExchangeContext.
+     *  @param[in]    aPayload             A payload that has read request data
+     *
+     *  @retval #Others If fails to process read request
+     *  @retval #CHIP_NO_ERROR On success.
+     *
+     */
+    CHIP_ERROR OnWriteRequest(Messaging::ExchangeContext * apExchangeContext, System::PacketBufferHandle aPayload);
+
+    /**
+     *  Send Write Response to initiator
+     *
+     *  @param[in]    aPayload             A payload that has read request data
```

Comment:
this whole comment seems to restate the code (except maybe payload that mentions 'read request data' instead of 'write request data' and I am unsure if that is correct or not.

Please update to only state things that are not obvious in the code.

Reply:
removed, thanks